### PR TITLE
Adds the blood-drunk miner as a potential mining "mini" boss

### PR DESCRIPTION
**Pull request by ChangelingRain**
It's very unique, very aggressive... VERY aggressive.
It dashes at you, fires a KA projectile, and attacks almost as rapidly as a player does in melee.

It does not currently appear since I want to use the tiles from #28760 in the ruin it shows up in.

- [x] Actual effect for crusher trophy, as if you can actually get the trophy off of it without SUPER DYING

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		Shoot(target)
+		changeNext_move(CLICK_CD_RANGE)
+	transform_weapon()
+
+/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter/proc/quick_attack_loop()
+	if(next_move <= world.time)
+		sleep(1)
+		.() //retry
+		return
+	sleep((next_move - world.time) * 1.5)
```
Comment (Iamgoofball) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: ahem

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	projectilesound = 'sound/weapons/kenetic_accel.ogg'
+	ranged = 1
+	pixel_x = -16
+	crusher_loot = list(/obj/item/weapon/melee/transforming/cleaving_saw, /obj/item/weapon/gun/energy/kinetic_accelerator, /obj/item/crusher_trophy/hunter_eye)
+	loot = list(/obj/item/weapon/melee/transforming/cleaving_saw, /obj/item/weapon/gun/energy/kinetic_accelerator)
+	wander = FALSE
+	del_on_death = TRUE
+	stat_attack = UNCONSCIOUS
+	blood_volume = BLOOD_VOLUME_NORMAL
+	var/obj/item/weapon/melee/transforming/cleaving_saw/hunter/CS
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: This var could use a better name imo

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	var/adjustment_amount = amount * 0.15
+	if(world.time + adjustment_amount > next_move)
+		changeNext_move(adjustment_amount) //attacking it interrupts it attacking, but only briefly
+	. = ..()
+
+/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter/death()
+	if(health > 0)
+		return
+	new /obj/effect/temp_visual/dir_setting/hunter_death(loc, dir)
+	..()
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: `return ..()`, it's a better practice!

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter/Move(atom/newloc)
+	if(dashing || (newloc && newloc.z == z && (istype(newloc, /turf/open/floor/plating/lava) || istype(newloc, /turf/open/chasm)))) //we're not stupid!
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter/ex_act(severity, target)
+	if(dash())
+		return
+	..()
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: same here.

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+	transform_weapon()
+	INVOKE_ASYNC(src, .proc/quick_attack_loop)
+	if(prob(10))
+		INVOKE_ASYNC(src, .proc/dash)
+	return TRUE
+
+/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter/do_attack_animation(atom/A, visual_effect_icon, obj/item/used_item, no_effect, end_pixel_y)
+	if(!used_item)
+		used_item = CS
+	..()
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: same

File: code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+		for(var/t in accessable_turfs)
+			accessable_turfs[t] = (accessable_turfs[t] - closest_dist) * 10
+		target_turf = pickweight(accessable_turfs)
+	else
+		target_turf = pick(accessable_turfs)
+	var/turf/step_back_turf = get_step(target_turf, get_cardinal_dir(target_turf, own_turf))
+	var/turf/step_forward_turf = get_step(own_turf, get_cardinal_dir(own_turf, target_turf))
+	new /obj/effect/temp_visual/small_smoke/halfsecond(step_back_turf)
+	new /obj/effect/temp_visual/small_smoke/halfsecond(step_forward_turf)
+	var/obj/effect/temp_visual/decoy/fading/halfsecond/D = new /obj/effect/temp_visual/decoy/fading/halfsecond(own_turf, src)
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/megafauna/blood_drunk_hunter.dm]: Redundant path after new

**Comment by Repukan** [comment]
>it uses the cap

That's how I know it's a scrub.


Why not make it look like the scavenger from RoR, and have it have (only visually) the effects from other lavaland drops?

**Comment by Kraseo** [comment]
he played too much minecraft

File: code/datums/status_effects/buffs.dm [hunk 7]
```diff
@@ -262,3 +262,113 @@
...
+	tick()
+	owner.maxHealth *= 0.1
+	owner.bruteloss *= 0.1
+	owner.fireloss *= 0.1
+	owner.toxloss *= 0.1
+	owner.oxyloss *= 0.1
+	owner.cloneloss *= 0.1
+	owner.staminaloss *= 0.1
+	owner.updatehealth()
+	add_logs(owner, null, "lost blood-drunk stun immunity")
```
Comment (lzimann) [inline code/datums/status_effects/buffs.dm]: You can use `owner.log_message("msg", INDIVIDUAL_ATTACK_LOG)` instead!

**Comment by Repukan** [comment]
Do they have a GPS signal too?

**Comment by GunHog** [comment]
It does not look like it, no.

**Comment by GunHog** [comment]
It does not look like it, no.

**Comment by Repukan** [comment]
Sounds !!FUN!!

**Comment by dannno** [comment]
I like this a lot but I think you could take more creative liberty with what you call the boss
"blood drunk hunter" is a little bit on the nose

**Comment by lzimann** [comment]
Yea, should probably add a GPS signal since it's not exactly an easy mob.

**Comment by ExcessiveUseOfCobblestone** [comment]
GPS: Slurred Signal

**Comment by DoctoreMad** [comment]
Logically the GPS signal should be like that of another miner's GPS.
(FUN)

File: code/datums/status_effects/buffs.dm [hunk 8]
```diff
@@ -70,7 +70,7 @@
 	return ..()
 
 /datum/status_effect/vanguard_shield/on_apply()
-	add_logs(owner, null, "gained Vanguard stun immunity")
+	owner.log_message("gained Vanguard stun immunity", INDIVIDUAL_ATTACK_LOG)
```
Comment (DoctoreMad) [inline code/datums/status_effects/buffs.dm]: Very atomic shit going on right there

File: code/datums/status_effects/buffs.dm [hunk 9]
```diff
@@ -262,3 +262,113 @@
...
+		owner.staminaloss *= 10
+		owner.updatehealth()
+		last_health = owner.health
+		last_bruteloss = owner.getBruteLoss()
+		last_fireloss = owner.getFireLoss()
+		last_toxloss = owner.getToxLoss()
+		last_oxyloss = owner.getOxyLoss()
+		last_cloneloss = owner.getCloneLoss()
+		last_staminaloss = owner.getStaminaLoss()
+		owner.log_message("gained blood-drunk stun stun immunity", INDIVIDUAL_ATTACK_LOG)
```
Comment (DoctoreMad) [inline code/datums/status_effects/buffs.dm]: stun stun immunity: For when you're afraid the first person to doublestun will win the match...

**Comment by DoctoreMad** [comment]
How do you get blood drunk status, anyways?

**Comment by Repukan** [comment]
Drink a lot of blood

**Comment by ChangelingRain** [comment]
Detonate a crusher mark with the crusher reward from this boss active.

**Comment by DoctoreMad** [comment]
But
But
But if I beat the boss I'm going to use his cleaver, not my Kinectic Crusher
[[crying in the background]]

**Comment by Armhulen** [comment]
retheme it, this one SUCKS!

**Comment by GunHog** [comment]
Yes, he looks like a bloody miner. Why not give him fancy hunter's garb complete with a big goofy hat?

**Comment by Hatfish** [comment]
Give them the inquisitor's hardsuit instead.

**Comment by Repukan** [comment]
Looks cool, fuck the non miner players that'll never interact with it

**Comment by improvedname** [comment]
Now im gonna die by the thing i thought was my friend miner

**Comment by Shadowlight213** [comment]
Inb4 people ahelp it

**Comment by dannno** [comment]
"loot-drunk miner"

?

**Comment by Armhulen** [comment]
"goes-for-tendrils-instead-of-mats-roundstart miner"

?

**Comment by dannno** [comment]
Loot-Crazed Miner

**Comment by Armhulen** [comment]
can we just have it not be a miner have it be like something more interesting

**Comment by DoctoreMad** [comment]
>Inb4 people ahelp it

Heh
What would be a good garb for him, anyways? Miner hardsuit since nobody uses that? Maybe put some "legion" bits on him to make him distinguishable? Or maybe go full with Armhulen's description

**Comment by Iamgoofball** [comment]
"Can we just scrap everything about it and make it a legion reskin"

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 10]
```diff
@@ -250,6 +250,19 @@
 	if(.)
 		H.charge_time += bonus_value
 
+//blood-drunk hunter
+/obj/item/crusher_trophy/hunter_eye
+	name = "eye of a blood-drunk hunter"
+	desc = "Its pupil is collapsed and turned to mush, indicating the onset of the scourge of beasts. Suitable as a trophy for a kinetic crusher."
```
Comment (KorPhaeron) [inline code/modules/mining/equipment/kinetic_crusher.dm]: Can we be a bit more subtle please

**Review by imtakingabreakdontatme — DISMISSED** [review]

File: code/modules/mining/equipment/kinetic_crusher.dm (hunk 10 above)
Comment (lzimann) [inline code/modules/mining/equipment/kinetic_crusher.dm]: Subtle about what exactly

Comment (KorPhaeron) [inline code/modules/mining/equipment/kinetic_crusher.dm]: http://bloodborne.wiki.fextralife.com/Eye+of+a+Blood-drunk+Hunter

**Comment by Chronitonity** [comment]
having the status effect be called blood drunk when you get it from using the crusher just sounds weird and trying to hard to be a BB reference

**Comment by dannno** [comment]
@Armhulen The fact that it's a miner is what makes it interesting and connects the reference

we already have a shitload of monsters and whatnot, my dude
a miner enemy IS interesting.

**Comment by ExcessiveUseOfCobblestone** [comment]
Think of it like the final boss of dark souls 3 where deep down if you played through the series you're actually fighting a mere shell of your former self.

**Comment by dannno** [comment]
or you can think of it in terms of a blood drunk hunter
from the superior game

**Review by lzimann — APPROVED** [review]
The code is fine

**Comment by DoctoreMad** [comment]
"If you put a legion head on his chest he'll just be a legion reskin reeeeeee I think players shouls suddenly die to other miners if they don't pre-emptively aim their KA on them to make sure they aren't the mining mini-boss"

**Comment by RemieRichards** [comment]
What happened to <i>Space Station 13 is a paranoia-laden round-based roleplaying game set against the backdrop of a nonsensical, metal death trap masquerading as a space station, with charming spritework designed to represent the sci-fi setting and its dangerous undertones. Have fun, and survive!</i>

P A RA N O I A

**Comment by RemieRichards** [comment]
Also it's like the source material for The Thing, 'Who Goes There?'
while I don't really care for the ham-fisted bloodbourne reference, I do like the "Can I trust them?" scenario.

**Comment by KorPhaeron** [comment]
Paranoia doesnt work with our ruleset

**Comment by RemieRichards** [comment]
>rules

it's a fucking NPC kor.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Yeah until you shoot your friend thinking its the monster and they shoot you back so you shoot them and then admins are asking you why you killed someone

**Comment by dannno** [comment]
add some effect or some kind of behaviour it does on sight that can easily confirm it as a bad guy and not a fellow miner

make it shriek or glow or do some shit idfk

**Comment by Repukan** [comment]
Make the player screen flash red when he enters the screen

**Comment by GunHog** [comment]
Just give him a fancy hunter's hat, geez.

**Comment by Iamgoofball** [comment]
make him the sniper from TF2

**Comment by RemieRichards** [comment]
@KorPhaeron >killing them rather than firing a warning shot or asking

Sounds like you're just a shit player tbh.

**Comment by lzimann** [comment]
@KorPhaeron decide on this already! open prs that are ready and not merged bothers me a bit!

**Comment by Iamgoofball** [comment]
It's good, let it in

**Comment by KorPhaeron** [comment]
Alright but no more bosses after this for a while, lavaland has enough high end enemies

**Comment by ChangelingRain** [comment]
@KorPhaeron tell me what you want already!

**Comment by Repukan** [comment]
Phrasing

**Comment by Nilons** [comment]
attacks much faster than a regular person
